core/server/master/src/main/java/alluxio/master/file/meta/LockedInodePath.java

```diff
@@ -273,6 +319,10 @@ public void addNextInode(Inode inode) {
     Preconditions.checkState(!fullPathExists());
     Preconditions.checkState(inode.getName().equals(mPathComponents[mLockList.numInodes()]));
 
+    // We need to flush the pending journals into the writer
```

Comment:
do you need to flush in removeLastInode() also?

Reply:
and also in traverse()

Reply:
removeLastInode() needs a flush. will update
traverse() doesn't need that because it actually acquires locks rather than releases/downgrade locks.